base/util/src/com/netscape/cmsutil/crypto/CryptoUtil.java

```diff
@@ -2748,6 +2750,47 @@ public static AlgorithmIdentifier getDefaultHashAlg()
         return hashAlg;
     }
 
+    /**
+    * importHmacSha1Key returns a key based on a byte array,
+    * which is originally a password. Used for the HMAC Digest algs.
+    *
+    * @param key the byte array representing the original password or secret.
+    * @return The JSS SymKey
+    *
+    */
+    public static Key importHmacSha1Key(byte[] key)
```

Comment:
I'm sorry, but I don't follow this or the purpose of it. Do you have a source for this?

AFAICT:

1. You generate a random AES-256 key (lines 2773 - 2775). 
2. You generate an empty/null IV (line 2777).
3. You construct the cipher in AES/CBC/PKCS5 mode (line 2781).
4. You pass the function parameter `key` to this cipher; since cipher in 3 was constructed with a random key (`wrappingKey`), this output (`wrappedKeyData`) should be random for multiple applications of the function on the same key.
5. You then unwrap the key (!?), returning the original key again? 

Is all of this a fancy way of converting a byte array to a `java.crypto.Key`? I think you might want to look at [`SecretKeySpec`](https://docs.oracle.com/javase/8/docs/api/index.html?javax/crypto/spec/SecretKeySpec.html) if so... :)

Reply:
Yeah. I got this code from a test in jss that tests out the JSS HMAC algs. The purpose for this method is to take a raw password / byte array and make a legal JSS key out of it so the HMAC can be executed properly.  The old HMADigest.java class did the manipulation of the password in java code, which caused us to be flagged.

Reply:
Wow that is terrible. I'll work on fixing this for the next JSS release.